diff --git a/ckanext/saml2auth/plugin.py b/ckanext/saml2auth/plugin.py
index 83f703b..d4c2345 100644
--- a/ckanext/saml2auth/plugin.py
+++ b/ckanext/saml2auth/plugin.py
@@ -28,7 +28,6 @@ import ckan.plugins.toolkit as toolkit
 from ckan.common import g
 import ckan.lib.base as base

-from ckanext.saml2auth.views.saml2auth import saml2auth
 from ckanext.saml2auth.cache import get_subject_id, get_saml_session_info
 from ckanext.saml2auth.spconfig import get_config as sp_config
 from ckanext.saml2auth import helpers as h
@@ -84,6 +83,7 @@ class Saml2AuthPlugin(plugins.SingletonPlugin):
     # IBlueprint

     def get_blueprint(self):
+        from ckanext.saml2auth.views.saml2auth import saml2auth
         return [saml2auth]

     # IConfigurer
@@ -95,20 +95,20 @@ class Saml2AuthPlugin(plugins.SingletonPlugin):

     # IAuthenticator

-    def logout(self):
+    # def logout(self):

-        response = _perform_slo()
+    #     response = _perform_slo()

-        if response:
-            domain = h.get_site_domain_for_cookie()
+    #     if response:
+    #         domain = h.get_site_domain_for_cookie()

-            # Clear auth cookie in the browser
-            response.set_cookie('auth_tkt', domain=domain, expires=0)
+    #         # Clear auth cookie in the browser
+    #         response.set_cookie('auth_tkt', domain=domain, expires=0)

-            # Clear session cookie in the browser
-            response.set_cookie('ckan', domain=domain, expires=0)
+    #         # Clear session cookie in the browser
+    #         response.set_cookie('ckan', domain=domain, expires=0)

-        return response
+    #     return response


 def _perform_slo():
