diff --git a/ckan/model/__init__.py b/ckan/model/__init__.py
index 03f3c13..9981737 100644
--- a/ckan/model/__init__.py
+++ b/ckan/model/__init__.py
@@ -266,7 +266,7 @@ class Repository():
         self.reset_alembic_output()
         alembic_config = AlembicConfig(self._alembic_ini)
         alembic_config.set_main_option(
-            "sqlalchemy.url", str(self.metadata.bind.url)
+            "sqlalchemy.url", config.get("sqlalchemy.url")
         )
         try:
             sqlalchemy_migrate_version = self.metadata.bind.execute(

diff --git a/ckanext/datastore/backend/postgres.py b/ckanext/datastore/backend/postgres.py
index 41369bbf8..716726408 100644
--- a/ckanext/datastore/backend/postgres.py
+++ b/ckanext/datastore/backend/postgres.py
@@ -1694,7 +1694,7 @@ class DatastorePostgresqlBackend(DatastoreBackend):
         '''
         write_connection = self._get_write_engine().connect()
         read_connection_user = sa_url.make_url(self.read_url).username
-
+        read_connection_user = read_connection_user.split('@')[0]
         drop_foo_sql = u'DROP TABLE IF EXISTS _foo'

         write_connection.execute(drop_foo_sql)
