diff --git a/ckanext/resourceproxy/blueprint.py b/ckanext/resourceproxy/blueprint.py
index 035b144c5..e968d66a7 100644
--- a/ckanext/resourceproxy/blueprint.py
+++ b/ckanext/resourceproxy/blueprint.py
@@ -68,7 +68,7 @@ def proxy_resource(context: Context, data_dict: DataDict):
         if not did_get:
             r = requests.get(url, timeout=timeout, stream=True)

-        response.headers[u'content-type'] = r.headers[u'content-type']
+        response.headers[u'content-type'] = r.headers.get(u'content-type')
         response.charset = r.encoding or "utf-8"

         length = 0